sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -40,6 +41,130 @@ def test_input_estimator_unchanged():
     assert_true(transformer.estimator is est)
 
 
+def check_invalid_max_features(est, X, y):
+    max_features = X.shape[1]
+    for invalid_max_n_feature in [-1, max_features + 1, 'gobbledigook']:
+        transformer = SelectFromModel(estimator=est,
+                                      max_features=invalid_max_n_feature,
+                                      threshold=-np.inf)
+        assert_raises(ValueError, transformer.fit, X, y)
+
+
+def check_valid_max_features(est, X, y):
+    max_features = X.shape[1]
+    for valid_max_n_feature in [0, max_features, 'all', 5]:
+        transformer = SelectFromModel(estimator=est,
+                                      max_features=valid_max_n_feature,
+                                      threshold=-np.inf)
+        X_new = transformer.fit_transform(X, y)
+        if valid_max_n_feature == 'all':
+            valid_max_n_feature = max_features
+        assert_equal(X_new.shape[1], valid_max_n_feature)
+
+
+class FixedImportanceEstimator(BaseEstimator):
+    def __init__(self, importances):
+        self.importances = importances
+
+    def fit(self, X, y=None):
+        self.feature_importances_ = np.array(self.importances)
+
+
+def check_max_features(est, X, y):
```

Comment:
I'd rather this be a separate `test_max_features`, even if it means generating X and y again.